### ISSUE TITLE
Developing a training pipeline to handle data training, testing, and evaluation and assessment of model performance with K-fold cross-validation

### ISSUE THREAD

**Issue by mxochicale** [oocular/ready#93]
### Motivation

* [x] Data samples are 20 images and labels from [mobius dataset within the github repo](https://github.com/oocular/ready/tree/main/data/mobious/sample-frames/test3000x1700) to accelerate model development. DONE for @mxochicale by creating branch 93-adding-extra-samples
* [ ] Splitting data into train, test and evaluation to develop an appropriate training pipeline. IN PROGRESS for @stoppedtime24 
  * [x] Step 1. Adding Training and validation into the main loop as in this blog https://stanford.edu/~shervine/blog/pytorch-how-to-generate-data-parallel. 
  * [x] Step 2. Adding evaluation with testset as in https://machinelearningmastery.com/training-a-pytorch-model-with-dataloader-and-dataset/
  * [x] Step 3. Loading trained model and evaluate model as shown here https://docs.pytorch.org/tutorials/beginner/saving_loading_models.html?highlight=eval
  * [ ] Review this reference https://debuggercafe.com/saving-and-loading-the-best-model-in-pytorch/
  * Other references: 
    * See how these functions def make_train_dataset, def make_val_dataset, def make_test_dataset are defined here:   https://www.kaggle.com/code/suchith0312/pytorch-dataloader-testing
    * [`train_dataset, test_dataset = torch.utils.data.random_split(full_dataset, [0.8, 0.2])`](https://stackoverflow.com/questions/50544730/how-do-i-split-a-custom-dataset-into-training-and-test-datasets)
  * This might be a clear example 
* [ ] Using K-fold cross validation to asses model performance. See this [reference](https://discuss.pytorch.org/t/using-k-fold-cross-validation-to-train-my-model/196288/2) for `ConcatDataset` and `sklearn.model_selection import KFold`
* [ ] Train model with A100 Nvidia GPU (80GB)
* [ ] Run inference in a local GPU-model (maybe not)?  

### Additional context

* Current [train_unet_with_mobious](https://github.com/oocular/ready/blob/main/src/ready/apis/train_mobious.py) use the whole dataset of 1143 samples of the dataset using [MobiousDataset](https://github.com/oocular/ready/blob/main/src/ready/utils/datasets.py)
* Using [cross validation](https://stackoverflow.com/questions/62954433/how-to-predict-with-the-test-dataset-while-using-cross-validation) for model performance assessment is one good initial step but also you can use it for model tuning (i.e. hyperparameter search).
* Reference describing k-fold-cross-validation:
  * https://www.aionlinecourse.com/tutorial/machine-learning/k-fold-cross-validation
  * https://knowledge.dataiku.com/latest/ml-analytics/ml-concepts/concept-model-validation.html
  * https://stackoverflow.com/questions/58391295/how-is-cross-validation-performed-and-how-gridsearchcv-specifically
  * https://www.mdpi.com/2072-4292/17/7/1256
  * https://www.frontiersin.org/journals/bioengineering-and-biotechnology/articles/10.3389/fbioe.2025.1580502/full
  * Ensembling, Blending & Stacking: https://www.youtube.com/watch?v=TuIgtitqJho

## Notes.
* ...
